patches/server/0860-More-Sign-Block-API.patch

```diff
@@ -1,48 +1,59 @@
 
 diff --git a/src/main/java/net/minecraft/world/level/block/entity/SignBlockEntity.java b/src/main/java/net/minecraft/world/level/block/entity/SignBlockEntity.java
-index bfe8029852385875af4ebe73c63e688f61042021..a28be7a332659be655f419d969e0c64e659b6c21 100644
+index bfe8029852385875af4ebe73c63e688f61042021..3070cd2b588f5a69fd8c0d3551e16251680d8c27 100644
 --- a/src/main/java/net/minecraft/world/level/block/entity/SignBlockEntity.java
 +++ b/src/main/java/net/minecraft/world/level/block/entity/SignBlockEntity.java
 @@ -68,12 +68,17 @@ public class SignBlockEntity extends BlockEntity implements CommandSource { // C
      }
  
      public boolean isFacingFrontText(net.minecraft.world.entity.player.Player player) {
-+        // Paper start - Add Sign#getInteractableSideFor
++        // Paper start - More Sign Block API
 +        return this.isFacingFrontText(player.getX(), player.getZ());
 +    }
 +    public boolean isFacingFrontText(double x, double z) {
-+        // Paper end - Add Sign#getInteractableSideFor
++        // Paper end - More Sign Block API
          Block block = this.getBlockState().getBlock();
  
          if (block instanceof SignBlock blocksign) {
              Vec3 vec3d = blocksign.getSignHitboxCenterPosition(this.getBlockState());
 -            double d0 = player.getX() - ((double) this.getBlockPos().getX() + vec3d.x);
 -            double d1 = player.getZ() - ((double) this.getBlockPos().getZ() + vec3d.z);
-+            double d0 = x - ((double) this.getBlockPos().getX() + vec3d.x); // Paper - Add Sign#getInteractableSideFor
-+            double d1 = z - ((double) this.getBlockPos().getZ() + vec3d.z); // Paper - Add Sign#getInteractableSideFor
++            double d0 = x - ((double) this.getBlockPos().getX() + vec3d.x); // Paper - More Sign Block API
++            double d1 = z - ((double) this.getBlockPos().getZ() + vec3d.z); // Paper - More Sign Block API
              float f = blocksign.getYRotationDegrees(this.getBlockState());
              float f1 = (float) (Mth.atan2(d1, d0) * 57.2957763671875D) - 90.0F;
  
 diff --git a/src/main/java/org/bukkit/craftbukkit/block/CraftSign.java b/src/main/java/org/bukkit/craftbukkit/block/CraftSign.java
-index 8303343ecca91076839f4436d6b3a3bf4739c2fd..cefbb015a77b9e3cab56e5ed4fe35fba91641632 100644
+index 8303343ecca91076839f4436d6b3a3bf4739c2fd..b9532a6ab6470e1af7509c16d42c1a98fdbee17d 100644
 --- a/src/main/java/org/bukkit/craftbukkit/block/CraftSign.java
 +++ b/src/main/java/org/bukkit/craftbukkit/block/CraftSign.java
-@@ -198,6 +198,14 @@ public class CraftSign<T extends SignBlockEntity> extends CraftBlockEntityState<
+@@ -198,6 +198,24 @@ public class CraftSign<T extends SignBlockEntity> extends CraftBlockEntityState<
      }
      // Paper end
  
-+    // Paper start - side facing API
++    // Paper start - More Sign Block API
++    @Override
++    public java.util.UUID getAllowedEditorUniqueId() {
++        return getTileEntity().getPlayerWhoMayEdit();
```

Comment:
and here?

Also, this is technically duplicate API with `getAllowedEditor`. Should we deprecate the old one here?